src/libraries/System.Text.Json/Common/JsonIgnoreCondition.cs

```diff
@@ -30,5 +30,15 @@ public enum JsonIgnoreCondition
         /// This is applied only to reference-type properties and fields.
         /// </summary>
         WhenWritingNull = 3,
+
+        /// <summary>
+        /// Property is ignored during serialization
+        /// </summary>
+        WhenWriting = 4,
+
+        /// <summary>
+        /// Property is ignored during deserialization
+        /// </summary>
+        WhenReading = 5,
```

Comment:
@eiriktsarpalis, how does someone specify multiple of these? e.g. if I have a property that I want to be ignored when reading and when writing null, how do I achieve that? Or is that not a scenario that matters?

Reply:
Not possible, which shows the limits of the `JsonIgnoreCondition` design. These enums eventually map to the `JsonPropertyInfo.ShouldSerialize` and `JsonPropertyInfo.Set` delegates, so a user wanting to introduce their own ignore conditions can configure these via the contract customization APIs.